src/Microsoft.AspNetCore.Razor.Evolution/Legacy/HtmlMarkupParser.cs

```diff
@@ -1706,7 +1718,7 @@ private int HandleNestingSequence(string sequence, int position, int currentNest
                 if (string.Equals(possibleStart, sequence, Comparison))
                 {
                     // Capture the current symbol and "put it back" (really we just want to clear CurrentSymbol)
-                    var bookmark = Context.Source.Position;
+                    var bookmark = Context.Source.Position - CurrentSymbol.Content.Length;
```

Comment:
This defines bookmark as the start the current symbol. I guess this could be just `CurrentStart`.